underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -186,12 +197,13 @@ public boolean delete(String path, boolean recursive) throws IOException {
 
   @Override
   public boolean exists(String path) throws IOException {
+    //TODO(adit): this hack should not be here
     // To get better performance Swift driver does not create a _temporary folder.
     // This optimization should be hidden from Spark, therefore exists _temporary will return true.
     if (path.endsWith("_temporary")) {
       return true;
     }
-    return getObject(path).exists();
+    return isRoot(path) || isFile(path) || isDirectory(path);
```

Comment:
exists checks for the existence of a file or a folder. i.e, given 'path', 2 calls to getObject are checking existence of 'path' or 'path/'. i don't see how this extra call can be avoided. 
this is consistent with s3 behavior (2 calls to getObjectDetails), just that the code is written a little different.